pkgs/development/python-modules/nanobind/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib, buildPythonPackage, pythonOlder, fetchFromGitHub, cmake, eigen, ninja
+, scikit-build, pytestCheckHook, numpy, scipy, torch, jax, jaxlib, tensorflow
+, setuptools }:
+buildPythonPackage rec {
+  pname = "nanobind";
+  version = "1.9.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "wjakob";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-6swDqw7sEYOawQbNWD8VfSQoi+9wjhOhOOwPPkahDas=";
+    fetchSubmodules = true;
+  };
+
+  disabled = pythonOlder "3.8";
+
+  nativeBuildInputs = [ cmake ninja scikit-build setuptools ];
+  buildInputs = [ eigen ];
+  dontUseCmakeBuildDir = true;
+
+  postBuild = ''
+    # build tests
```

Comment:
Or should it be `preCheck`? Since it's only for the tests.

Reply:
Indeed, thanks for the comment.